indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -178,14 +183,25 @@ public final TaskStatus run(TaskToolbox taskToolbox) throws Exception
       throw e;
     }
     finally {
-      cleanUp(taskToolbox, taskStatus);
+      try {
+        cleanUp(taskToolbox, taskStatus);
+      }
+      finally {
+        cleanupCompletionLatch.countDown();
+      }
     }
   }
 
   public abstract TaskStatus runTask(TaskToolbox taskToolbox) throws Exception;
 
+  @Override
   public void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
   {
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
I think this might not be needed. You could just use `Thread.interrupted` to have the same effect.

Reply:
updated.